app/src/main/java/com/example/reduxtestapp/data/repository/country/CountryRepositoryImpl.kt

```diff
@@ -57,4 +62,84 @@ class CountryRepositoryImpl (
         }
     }
 
+    override suspend fun searchByLanguageAndCurrency(
+        language: String,
+        currency: String
+    ): Either<ErrorState, List<CountryDto>> {
+        val deferredLanguageResult = CoroutineScope(Dispatchers.IO).async {
+            searchByLanguage(language)
+        }
+        val deferredCurrencyResult = CoroutineScope(Dispatchers.IO).async {
+            searchByCurrency(currency)
+        }
+
+        val languageResult = deferredLanguageResult.await()
+        val currencyResult = deferredCurrencyResult.await()
+
+        return Either.zipOrAccumulate(languageResult, currencyResult) { langList, currList ->
+            langList.intersect(currList.toSet()).toList()
+        }.mapLeft { errorList ->
+            if (errorList.all { it is ErrorState.EmptyListError })
+                ErrorState.EmptyListError
+            else ErrorState.CountriesError(null)
+        }
```

Comment:
look up `either { }` block and `.bind()` method on either and see if you can use that instead of zip